projects/igniteui-angular/src/lib/services/overlay/overlay.spec.ts

```diff
@@ -188,6 +188,80 @@ describe('igxOverlay', () => {
         UIInteractions.clearOverlay();
     });
 
+    describe('Pure Unit Test', () => {
+        configureTestSuite();
+        const mockElement: any = {
+            style: { visibility: '', cursor: '', transitionDuration: '' },
+            classList: { add: () => { }, remove: () => { } },
+            appendChild: () => { },
+            removeChild: () => { },
+            addEventListener: (type: string, listener: (this: HTMLElement, ev: MouseEvent) => any) => { },
+            removeEventListener: (type: string, listener: (this: HTMLElement, ev: MouseEvent) => any) => { },
+            getBoundingClientRect: () => ({ width: 10, height: 10 }),
+            insertBefore: (newChild: HTMLDivElement, refChild: Node) => { },
+            contains: () => { }
+        };
+        mockElement.parent = mockElement;
+        mockElement.parentElement = mockElement;
+        const mockElementRef: any = { nativeElement: mockElement };
+        const mockFactoryResolver: any = {
+            resolveComponentFactory: (c: any) => {
+                return {
+                    create: (i: any) => {
+                        return {
+                            hostView: '',
+                            location: mockElementRef,
+                            changeDetectorRef: { detectChanges: () => { } },
+                            destroy: () => { }
+                        };
+                    }
+                };
+            }
+        };
+        const mockApplicationRef: any = { attachView: (h: any) => { }, detachView: (h: any) => { } };
+        const mockInjector: any = {};
+        const mockAnimationBuilder: any = {};
+        const mockDocument = {
+            body: mockElement,
+            defaultView: mockElement,
+            createElement: () => mockElement,
+            appendChild: () => { },
+            addEventListener: (type: string, listener: (this: HTMLElement, ev: MouseEvent) => any) => { },
+            removeEventListener: (type: string, listener: (this: HTMLElement, ev: MouseEvent) => any) => { }
+        };
+        const mockNgZone: any = {};
+        const mockPlatformUtil: any = { isIOS: false };
+
+        const overlay = new IgxOverlayService(
```

Comment:
This config is great! I think it should be moved in a `beforeEach` statement - in case we end up adding more unit tests and don't want to take a chance on the overlay instance being "polluted".